5 establishing connection with database

hubertborysowski opened the pull request:
# Read me
Establishment with database works fine.

`For tests only` sections are still there, so you can check if everything works fine on your devices.
Health check is made by `catch` method.

**NOTE:** The program seeks database called `test` with table `students` in it. Change it to desired one or create temporary database with specified table (of course remember to fill this table with some data). Program seeks for int, string, and int again.

hubertborysowski committed:
5 connection to database, java error encountered

hubertborysowski committed:
5 pom fixes, java error

Fixes on new dependency in pom.xml
Java program with connection establishment doesn't work due to unexpected error with syntax.

hubertborysowski committed:
5 establishing connection with database | working

hubertborysowski committed:
Merge branch '5-establishing-connection-with-database' of https://github.com/Mateusz-Dobrzynski/meals-management-system into 5-establishing-connection-with-database

hubertborysowski committed:
5 establishing connection with database | works

Everything works fine. Health check is performed from `catch` method.

hubertborysowski committed:
5 establishing connection with database | fixes

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java:
```diff
@@ -0,0 +1,30 @@
+import java.sql.Statement;
+
+public class MysqlConnection {
+    public void sqlConnection() {
```
This method should be documented. Other than that, the name is not self-explanatory so I'm not exactly sure what is it that this method is doing

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java:
```diff
@@ -0,0 +1,30 @@
+            Connection connection = DriverManager.getConnection (url, username, password);
+            Statement statement = connection.createStatement ();
+            ResultSet resultSet = statement.executeQuery ("select * from students"); /* Change `students` to desired table. */
+            /* For tests only. */
+            while (resultSet.next ()) {
+                System.out.println(
+                    resultSet.getInt (1)+" "+resultSet.getString (2)+" "+resultSet.getInt (3));
+            }
```
Instead of including tests in the source code, create separate tests for that. In order to do that, you need to mock the behavior of the database to see if the application behaves as expected depending on whether the DB is working properly or not.

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java:
```diff
@@ -0,0 +1,30 @@
+            connection.close ();
+        }
+        catch (Exception exception) {
+            System.out.println ("An error has occured.");
```
Avoid using magic values. Use a static variable at the beginning of the file or remove this line altogether.

hubertborysowski replied:
It is impossible not to use catch function in this form. Even [official documentation](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-usagenotes-connect-drivermanager.html) shows this kind of practice.

hubertborysowski replied:
SQL demands usage of this practice as the connection establishment enforces that.

Mateusz-Dobrzynski replied:
My comment does not refer to any of the above but to the magic value you used. Refrain from using them, if possible.

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java:
```diff
@@ -0,0 +1,30 @@
+            Class.forName ("com.mysql.cj.jdbc.Driver");
+            Connection connection = DriverManager.getConnection (url, username, password);
+            Statement statement = connection.createStatement ();
+            ResultSet resultSet = statement.executeQuery ("select * from students"); /* Change `students` to desired table. */
```
Avoid using magic values

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java:
```diff
@@ -0,0 +1,30 @@
+
+public class MysqlConnection {
+    public void sqlConnection() {
+        String url = "jdbc:mysql://localhost:3306/test"; /* Change `test` to desired database.*/
```
Avoid using magic values. Use a static variable at the beginning of the file

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java:
```diff
@@ -0,0 +1,30 @@
+            /* For tests only. */
+            while (resultSet.next ()) {
+                System.out.println(
+                    resultSet.getInt (1)+" "+resultSet.getString (2)+" "+resultSet.getInt (3));
```
Avoid using magic values. This line is not readable, nor self-explanatory. Looking at it, it is not possible to determine, what data will be printed. Try to make it cleaner, e.g. by assigning elements of the `resultSet` to separate variables. Other than that, are you sure that this line will work if the query will be changed?

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplication.java:
```diff
@@ -3,17 +3,20 @@
-		SpringApplication.run(MealsManagementSystemApplication.class, args);
+	public static void main (String[] args) {
+		SpringApplication.run (MealsManagementSystemApplication.class, args);
+		/* 
+		 * For tests only. 
+		 * TO-DO: Change to desired location when the app logic will be ready. 
+		 */
```
This class handles the app logic. Depending on what the method is meant to be doing (it is not possible to figure this out by looking at it), it might be at a right spot

Mateusz-Dobrzynski reviewed (changes requested):
You need to clean up and document your code, as well as create tests for it

hubertborysowski committed:
5 establishing connection with database

Documentation added. Unnecessary lines removed. Naming changes.

hubertborysowski committed:
5 establishing connection with database

Documentation added. Unnecessary lines removed. Naming changes.

Mateusz-Dobrzynski commented on src/test/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplicationTests.java:
```diff
@@ -8,6 +8,8 @@ class MealsManagementSystemApplicationTests {
 
 	@Test
 	void contextLoads() {
+		EstablishConnectionWithDatabase mysqlConnectionObject = new EstablishConnectionWithDatabase () ;
+		mysqlConnectionObject.createSqlConnection ();
```
This test does not make any assertions and as such doesn't test anything. Check several test cases. Mock the database behavior if you have to.

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java:
```diff
@@ -0,0 +1,62 @@
+    /**
+     * Method establishing connection with database.
+     *
+     * <p><code>accessConnection = DriverManager.getConnection (accessData.url, accessData.username, accessData.password);</code> connects to the database using provided data.</p>
+     * <p><code>executeSqlStatement = accessConnection.createStatement ();</code> creates an SQL statement, so we can prepare the object to receive data.</p>
+     * <p><code>resultSet = executeSqlStatement.executeQuery (databaseData.fooTable);</code> receives and stores information about the current table.</p>
```
You don't have to explain what the methods you're using do. They are documented by their authors. What matters is the documentation of your own code and what exactly it does.

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java:
```diff
@@ -0,0 +1,62 @@
+ */
+public class EstablishConnectionWithDatabase {
+    /**
+     * Method establishing connection with database.
```
This comment is redundant as the name of the method speaks for itself. Try to describe its behavior in more detail.

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java:
```diff
@@ -0,0 +1,62 @@
+            accessConnection = DriverManager.getConnection (accessData.url, accessData.username, accessData.password);
+            executeSqlStatement = accessConnection.createStatement ();
+            storeTableDataSet = executeSqlStatement.executeQuery (databaseData.fridges);
+            while(storeTableDataSet.next ()) {
+                /* Insert operations here. */
+            }
+            storeTableDataSet = executeSqlStatement.executeQuery (databaseData.productInstance);
+            while(storeTableDataSet.next ()) {
+                /* Insert operations here. */
+            }
+            storeTableDataSet = executeSqlStatement.executeQuery (databaseData.users);
+            while(storeTableDataSet.next ()) {
+                /* Insert operations here. */
+            }
+            storeTableDataSet = executeSqlStatement.executeQuery (databaseData.MeasurementUnits);
+            while(storeTableDataSet.next ()) {
+                /* Insert operations here. */
+            }
```
This code doesn't seem po perform any particular function and as such shouldn't be included in the project. You can move it to the test class if you insert the said `operations`

Mateusz-Dobrzynski reviewed (changes requested):
Do not include the generated documentation just yet. What matters are the comments in the source code (which, incidentally speaking, could be more detailed). Including the generated docs right now will make code review more difficult, not easier.

hubertborysowski committed:
5 establishing connection with database

Fixes.

hubertborysowski committed:
5 establishing connection with database | generated doc removal

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java:
```diff
@@ -0,0 +1,30 @@
+    public void sqlConnection() {
+        String url = "jdbc:mysql://localhost:3306/test"; /* Change `test` to desired database.*/
+        String username = "root";
+        String password = "";
```
Do not include credentials in the source code! Create a separate config file for that and DO NOT include its contents in the repository

Mateusz-Dobrzynski commented on src/test/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabaseTest.java:
This unit test is device-dependent (it requires a functional database) and as such does not serve its purpose. Other than that, it does not make any assertions. As stated in my previous reviews, create several test cases and mock the database behavior.

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java:
```diff
@@ -0,0 +1,37 @@
+import java.sql.Connection;
+import java.sql.DriverManager;
+class accessData {
+    static String url = "jdbc:mysql://localhost:3306/mms_database";
+    static String username = "root";
+    static String password = "";
+    static String connectionInstanceName = "com.mysql.cj.jdbc.Driver";
```
Avoid using magic values and **under no circumstances** include credentials in the source code. Create a separate config file for that and **do not** include it in your commits (include the file in `.gitignore`. Include instructions in the README file on where to enter the credentials.

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java:
```diff
@@ -0,0 +1,37 @@
+ */
+public class EstablishConnectionWithDatabase {
+    /* Database Data */
+    static String fridges = "select * from fridges";
+    static String productInstance = "select * from productinstance";
+    static String users = "select * from users";
+    static String MeasurementUnits = "select * from MeasurementUnits";
```
Unused variables

Mateusz-Dobrzynski commented on src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java:
```diff
@@ -0,0 +1,37 @@
+    public void createSqlConnection () {
+        Connection accessConnection;
+        try {
+            Class.forName (accessData.connectionInstanceName);
+            accessConnection = DriverManager.getConnection (accessData.url, accessData.username, accessData.password);
+            accessConnection.close ();
```
Unused connection. It is immediately closed and as such does not serve the application

hubertborysowski replied:
So you want this to receive (save in some kind of an array) and send data to the database?

Mateusz-Dobrzynski commented on src/test/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplicationTests.java:
This file is redundant. Turn `EstablishConnectionWithDatabaseTest.java` into a useful test instead.

hubertborysowski committed:
Merge branch 'dev' into 5-establishing-connection-with-database

hubertborysowski committed:
5 establish connection with database

- Added information in README
- Unused connection still present;
- Unused variables still present;
- Usless test still useless;

hubertborysowski commented:
Unused connection is still present, same as variables, waiting for response for previous comment about that.
Useless test is still useless. Trying to work on it, but having problems with mocking database.

hubertborysowski committed:
5 establishing connection with database

README fixes

hubertborysowski committed:
5 establishing connection with database

Minor changes. NOT READY FOR MERGING!

hubertborysowski committed:
Merge branch '5-establishing-connection-with-database' of https://github.com/Mateusz-Dobrzynski/meals-management-system into 5-establishing-connection-with-database

hubertborysowski commented:
Not ready for merging.
WIP.

hubertborysowski committed:
5 establish conneciton with database

JSON configuration file added.
Java file reads data and establishes connection with database.

TO-DO:
Mock database and create viable unit test.

hubertborysowski committed:
5 establishing connection with database

.gitignore update

hubertborysowski committed:
5 establishing connection with database

Mocking database works. Unit tests are working aswell.

hubertborysowski commented:
Works! 🤖
Waiting for review 📑

hubertborysowski committed:
5 establishing connection with database

Magic values fixes 🔮

hubertborysowski committed:
5 establishing connection with database

README fixes 📜

Mateusz-Dobrzynski commented on src/test/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplicationTests.java:
```diff
@@ -8,6 +8,12 @@ class MealsManagementSystemApplicationTests {
 
 	@Test
 	void contextLoads() {
+		EstablishConnectionWithDatabaseTest 
+		connectionWithDatabaseFileReferenceSetUp = new EstablishConnectionWithDatabaseTest () ;
+	connectionWithDatabaseFileReferenceSetUp.setUp ();
+		EstablishConnectionWithDatabaseTest
+		connectionWithDatabaseFileReference = new EstablishConnectionWithDatabaseTest () ;
+	connectionWithDatabaseFileReference.testMockDBConnection ();
```
You don't need to invoke the tests in a separate class. With these lines you test the same code twice: once directly in `EstablishConnectionWithDatabaseTest` and here again, invoking the same test methods. You can delete these lines.

Mateusz-Dobrzynski commented on src/test/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabaseTest.java:
```diff
@@ -0,0 +1,35 @@
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+ 
+public class EstablishConnectionWithDatabaseTest {
```
Document this class by explaining what it does

Mateusz-Dobrzynski reviewed (approved):
With mocks present, the code is tested and as such can be merged. Consider other comments before merging
